sdk/iot/hub/samples/src/sample_base64.c

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <openssl/bio.h>
+#include <openssl/buffer.h>
+#include <openssl/evp.h>
+#include <openssl/hmac.h>
+
+#include "az_span.h"
+#include "sample_base64.h"
+
+// Decode an input span from base64 to bytes
+az_result sample_base64_decode(az_span base64_encoded, az_span in_span, az_span* out_span)
+{
+  az_result result;
+
+  BIO* b64_decoder;
+  BIO* source_mem_bio;
+
+  memset(az_span_ptr(in_span), 0, (size_t)az_span_size(in_span));
+
+  // Create a BIO filter to process the bytes
+  b64_decoder = BIO_new(BIO_f_base64());
```

Comment:
>b64_decoder [](start = 2, length = 11)

Should check for null.

Reply:
we are not doing any checks for error, in any API. 

---
In reply to: [436091701](https://github.com/Azure/azure-sdk-for-c/pull/808#discussion_r436091701) [](ancestors = 436091701)

Reply:
updating